Asterix/tests/test_propagations.py

```diff
@@ -8,17 +8,26 @@ def test_mft_centering():
     """
     pdim = 8
     rad = pdim / 2
-    pup = roundpupil(pdim, rad)
+    pup = roundpupil(pdim, rad, center_pos='b')
 
     samp = 4
-    efield = mft(pup, real_dim_input=pdim, dim_output=pdim, nbres=samp)
+    efield = mft(pup, real_dim_input=pdim, dim_output=pdim, nbres=samp, dtype_complex='complex128')
```

Comment:
The pixel-comparison test still passes with `complex64`, the test with `np.transpose()` and `np.flip()` doesn't. This supports my idea to keep the pixel-comparison test, we can always add other tests on top, like you did. They simply test for different details.

Reply:
No I disagree. You got lucky, it seems that the 4 center pixels seems to be indeed exactly equal in this case. But you have the numerical error appearing when you test for other symetrical pixels : 

`print( (img[2, 3] == img[3, 2]))
print( (img[2, 4] == img[4, 2]))

print(img[2, 3], img[3, 2], img[2, 3] - img[3, 2])
print(img[2, 4], img[4, 2], img[2, 4] - img[4, 2])`

yields: 

```
False
False
415.76685519736236 415.7668551973622 1.7053025658242404e-13
415.76685519736236 415.7668551973622 1.7053025658242404e-13
```
These 2 pairs of pixels that should be exactly equals actually differ at the 13th decimal. 

If we agree that the problem come from the numerical noise of the MFT (and I think this last test proves it, the PSF is not symmetrical at the 13th decimal), then testing on exact equality of a single pixel is dangerous because we might not get lucky next time, those pixels can differ from very small value. The right test is a test using np.testing.assert_array_almost_equal either for pixel value or for transpose. I think that the second one (almost equality of transpose) is better because it test for all pixels : it is a more general test and actually include the first one,

Reply:
Lol
```python
In [6]: print( (img[2, 3] == img[3, 2]))
   ...: print( (img[2, 4] == img[4, 2]))
   ...: print(img[2, 3], img[3, 2], img[2, 3] - img[3, 2])
   ...: print(img[2, 4], img[4, 2], img[2, 4] - img[4, 2])
```
yields for me:
```bash
True
True
415.76685519736236 415.76685519736236 0.0
415.76685519736236 415.76685519736236 0.0
```

Reply:
ahah, more reason to avoid testing on the exact equality of pixels. ok let's settle this debate :-D 

```python
for i in range(pdim):
    for j in range(pdim):
        print(i,j,(img[i, j] == img[j, i]), img[i, j], img[j, i], img[i, j] - img[j, i])
```

Reply:
Some 1e-6 differences for complex64, some 1e-14 and 1e-15 errors for complex128. I vote for https://github.com/johanmazoyer/Asterix/pull/97#pullrequestreview-1168308255.

Reply:
I'll open an issue on the complex128/complex64, we need to discuss this

Reply:
but default is now complex128, we need to be safe more than fast right now

Reply:
Do you agree to put atol at 1e-10 then ? I feel this a good contraint but taht allow to variation between computers

Reply:
Yes, I would like to increase this a little but for now it's fine. How about you put this question in the new issue too?